src/arrow_writer.rs

```diff
@@ -0,0 +1,318 @@
+use std::io::{Seek, Write};
+
+use arrow::{
+    array::RecordBatch,
+    datatypes::{DataType as ArrowDataType, SchemaRef},
+};
+use prost::Message;
+use snafu::{ensure, ResultExt};
+
+use crate::{
+    error::{IoSnafu, Result, UnexpectedSnafu},
+    proto,
+    writer::{
+        column::EstimateMemory,
+        stripe::{StripeInformation, StripeWriter},
+    },
+};
+
+/// Construct an [`ArrowWriter`] to encode [`RecordBatch`]es into a single
+/// ORC file.
+pub struct ArrowWriterBuilder<W> {
+    writer: W,
+    schema: SchemaRef,
+    batch_size: usize,
+    stripe_byte_size: usize,
```

Comment:
What do you think using `stripe_size` or `stripe_length` for short?

Reply:
I like keeping the unit as part of the name, to be unambiguous